cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -256,8 +259,11 @@ impl<State> CoreEnvironment<State> {
 
         let service_config_path = maybe_make_service_config_file(flox, &lockfile)?;
 
-        let outputs =
-            BuildEnvNix.build(&flox.catalog_client, &lockfile_path, service_config_path)?;
+        let outputs = BuildEnvNix::new(flox.temp_dir.join(TEMPROOTS_DIR_NAME)).build(
```

Comment:
followup: TIL I have 7000 entries in ~/.cache/flox/process. Idk how we're leaking temp dirs but if we start putting gc roots in there it will be more of a problem. Seems like we need to store in `/tmp` or fix the leak or remove old directories in `flox gc`